src/plugin.ts

```diff
@@ -165,10 +169,10 @@ const plugin: FastifyPluginAsync<GraaspFileItemOptions> = async (fastify, option
   fastify.get<{ Params: IdParam }>('/:id/download', { schema: downloadSchema }, async (request, reply) => {
     const { member, params: { id }, log } = request;
 
-    const t1 = taskManager.createGetTaskSequence(member, id);
-    const t2 = new GetFileFromItemTask(member)
-    t2.getInput = () => ({ reply, path: storageRootPath, item: t1[0].result as Item<FileItemExtra> })
-    return runner.runSingleSequence([...t1, t2], log)
+    const item = await runner.runSingleSequence(await options.downloadValidation(id, member), log) as Item<FileItemExtra>;
+
+    const getFileTask = new GetFileFromItemTask(member, { reply, path: storageRootPath, item });
+    return runner.runSingleSequence([getFileTask], log);
```

Comment:
Wouldn't it possible to run once `runSingleSequence` with `downloadValidation` tasks and getFileTask?

Reply:
probably, but I haven't looked into it